package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -816,20 +816,21 @@ def _update_selection_2(self):
         self._s2_donors = {}
         self._s2_donors_h = {}
         self._s2_acceptors = {}
-        if self.selection1_type in ('donor', 'both'):
-            self._s2_acceptors = self._s2.select_atoms(
-                'name {0}'.format(' '.join(self.acceptors)))
-            self.logger_debug("Selection 2 acceptors: {0:d}".format(len(self._s2_acceptors)))
-        if self.selection1_type in ('acceptor', 'both'):
-            self._s2_donors = self._s2.select_atoms(
-                'name {0}'.format(' '.join(self.donors)))
-            self._s2_donors_h = {}
-            for i, d in enumerate(self._s2_donors):
-                tmp = self._get_bonded_hydrogens(d)
-                if tmp:
-                    self._s2_donors_h[i] = tmp
-            self.logger_debug("Selection 2 donors: {0:d}".format(len(self._s2_donors)))
-            self.logger_debug("Selection 2 donor hydrogens: {0:d}".format(len(self._s2_donors_h)))
+        if self._s2:
```

Comment:
Don't you need a similar check for `_s1`? 

You can probably re-use your test case by just switching selections.